### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by litvintech**
- Improved web3 dapp
- Implemented post Registry initialization with EntryCore
- Implemented divided logic for Entry creation/setting data
- Implemented versioning for Registry Creator
- Improved policies/security
- Implemented funding for Registries/Entries
- Added test coverage + report
- Added docs + contracts pipeline
- Refactoring

### CODE REVIEW

**Comment by litvintech** [comment]
after review and merge going to set v0.1 tag to master

**Comment by litvintech** [comment]
@xhipster @hleb-albau @Pechalka waiting for your review guys

File: contracts/registry/RegistryPermissionControl.sol [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        public
+        view
+        returns (
+            address
+        )
+    {
+        return admin;
+    }
+    
+    function getEntryPermissions()
```
Comment (hleb-albau) [inline contracts/registry/RegistryPermissionControl.sol]: fix getter name

File: contracts/common/RegistrySafe.sol [hunk 2]
```diff
@@ -1,29 +1,60 @@
...
-import "zeppelin-solidity/contracts/ownership/Ownable.sol";
-import "zeppelin-solidity/contracts/lifecycle/Destructible.sol";
 
+/**
+* @title RegistySafe contract
+* @author cyber•Congress
+* @dev Allows store etheirs and claim them by owner
+* @notice not recommend to use before release!
+*/
+contract RegistrySafe {
```
Comment (hleb-albau) [inline contracts/common/RegistrySafe.sol]: Maybe rename, not including `Registry` in name part? A bit confusing.

Comment (hleb-albau) [reply contracts/common/RegistrySafe.sol]: is Ownable?

File: test/common/AdminableTests.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
+const chai = require("chai")
+chai.should()
+chai.use(require("chai-bignumber")())
+chai.use(require("chai-as-promised"))
+
+const AdminableTestContract = artifacts.require('AdminableTestContract');
+
+contract('AdminableTestContract', function (accounts) {
```
Comment (hleb-albau) [inline test/common/AdminableTests.js]: Remove, no more `Adminable`

File: README.md [hunk 4]
```diff
@@ -1,9 +1,8 @@
 <h1 align="center">
-  <img src="https://raw.githubusercontent.com/cybercongress/chaingear/ERC721-integration/chaingear.png"
+  <img src="/logo_chaigear_970.png"
   alt="chaingear" width="970"></a>
 </h1>
 
-
 <h3 align="center">Most expensive Registry</h3>
```
Comment (SaveTheAles) [inline README.md]: Most expensive Registry isn't correct. I propose edit "most" to "the most"

File: README.md [hunk 5]
```diff
@@ -37,207 +39,60 @@ Entry type is defined during creation, so you can put into entry any custom logi
...
+4. Simple smart-contract EntryCore code generation in client
+5. Registries ABI and metainformation savings in IPFS
 
-chaingear.registerRegistry([], [], "BlockchainRegistry", "BLCHR", "", EntryCore.bytecode, {value: buildingFee, gas: 10000000})
+## [Contracts Overview](https://cybersearch.io/Chaingear/contracts/)
 
-```
+## [Configuring and deploying](https://cybersearch.io/Chaingear/development/)
 
+## General Chaingear/Registry ipeline
```
Comment (SaveTheAles) [inline README.md]: Edit ipline to pipeline

**Comment by litvintech** [comment]
@SaveTheAles @xhipster @hleb-albau 
1. refactored all small issues, which you described above
2. added check for entry zero balance before deletion
3. added symbol to return values of registryInfo, refactored registryInfo method
4. added method which allows changing of registry name
5. fixed auth check for entry updates
6. added the example for a unique field to EntryCore
7. increased tests coverage

**Pull request merged**